core/src/main/java/org/apache/calcite/rel/metadata/RelMdColumnUniqueness.java

```diff
@@ -67,6 +70,9 @@ public class RelMdColumnUniqueness
   public static final RelMetadataProvider SOURCE =
       ReflectiveRelMetadataProvider.reflectiveSource(
           new RelMdColumnUniqueness(), BuiltInMetadata.ColumnUniqueness.Handler.class);
+  static final Set<SqlKind> PASSTHROUGH_AGGREGATIONS =
```

Comment:
If a row has a unique column, then an aggregation that returns a value from that column is also to be unique. These are the aggregations that do that,

Reply:
Also consider making this an interface, similar to `SqlStaticAggFunction`, `SqlSingletonAggFunction`. That way, user-defined aggregate functions can have this property.